### PR TITLE
PMP: remove small connected components

### PR DESCRIPTION
**Pull request by MaelRL**
## Summary of Changes

Introduces a function to get rid of small connected components in terms of area / volume.

## Release Management

* Affected package(s): `Polygon_mesh_processing`
* Issue(s) solved (if any): --
* Feature/Small Feature (if any): [Remove_CCs_of_negligible_size](https://cgal.geometryfactory.com/CGAL/Members/wiki/Features/Small_Features/Remove_CCs_of_negligible_size) 
  * ~**pre-approval date:** 2019/10/28~
  * **pre-approval date:** 2019/11/12
* License and copyright ownership: no change

### CODE REVIEW

**Comment by MaelRL** [comment]
@sloriot: The main point I have not settled on yet is whether the area/volume should be of the total area / volume of the mesh, or of something else (e.g. the bbox's). 

The first mesh that I took to test actually had a gigantic open CC (big area, zero volume) and a few very tiny volumes that have to be removed. The function does clean the small CCs, because the area is small. However, imagine the case of large, flat CCs, then the area-based check would not delete these CCs (they have a large enough area), and the volume-based cleaning would not have cleaned anything either since the total volume is small.

**Comment by sloriot** [comment]
I think area and/or volume should be offered as np because in some cases you'll have open CC so volume will not be an option (same for tiny volumes with non tiny area)

File: Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/repair.h [hunk 1]
```diff
@@ -62,8 +64,241 @@
...
+  FaceIndexMap fim = choose_parameter(get_parameter(np, internal_np::face_index),
+                                      get_property_map(boost::face_index, tmesh));
+
+  FT area_threshold = choose_parameter(get_parameter(np, internal_np::area_threshold), -1.);
+  FT volume_threshold = choose_parameter(get_parameter(np, internal_np::volume_threshold), -1.);
+
+  // If no threshold is provided, compute it as a % of the bbox
+  const bool is_default_area_threshold = is_default_parameter(get_parameter(np, internal_np::area_threshold));
+  const bool is_default_volume_threshold = is_default_parameter(get_parameter(np, internal_np::volume_threshold));
+  const bool dry_run = choose_parameter(get_parameter(np, internal_np::dry_run),
```
Comment (MaelRL) [inline Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/repair.h]: Please indent properly and respect the existing style: single line.

File: Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/repair.h [hunk 2]
```diff
@@ -62,8 +64,241 @@
...
+    volume_threshold = CGAL::square(threshold_value);
+
+  // Compute the connected components only once
+  boost::vector_property_map<std::size_t, FaceIndexMap> face_cc(fim);
+  std::size_t num = connected_components(tmesh, face_cc, np);
+
+#ifdef CGAL_PMP_DEBUG_SMALL_CC_REMOVAL
+  std::cout << num << " different connected components" << std::endl;
+#endif
+  if(!dry_run){
```
Comment (MaelRL) [inline Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/repair.h]: no { (yourself did not use any on the previous call)

File: Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/repair.h [hunk 3]
```diff
@@ -58,8 +60,287 @@
...
+///
+/// Property maps for `CGAL::face_index_t` and `CGAL::vertex_index_t`
+/// must be either available as internal property maps
+/// to `tmesh` or provided as \ref pmp_namedparameters "Named Parameters".
+///
+/// \tparam TriangleMesh a model of `FaceListGraph` and `MutableFaceGraph`
+/// \tparam NamedParameters a sequence of \ref pmp_namedparameters "Named Parameters"
+///
+/// \param tmesh the triangulated polygon mesh
+/// \param nb_components_to_keep the number of components to be kept
```
Comment (lrineau) [inline Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/repair.h]: Doxygen error:
> include/CGAL/Polygon_mesh_processing/repair.h:99: warning: argument 'nb_components_to_keep' of command @param is not found in the argument list of CGAL::Polygon_mesh_processing::remove_connected_components_of_negligible_size(TriangleMesh &tmesh, const NamedParameters &np)

https://cgal.geometryfactory.com/CGAL/Manual_doxygen_test/CGAL-5.1-Ic-24/logs_master/Polygon_mesh_processing.log

**Comment by maxGimeno** [comment]
Successfully tested in https://cgal.geometryfactory.com/CGAL/testsuite/results-5.1-Ic-53.shtml

**Pull request merged**